fvm/evm/evm_test.go

```diff
@@ -3060,6 +3062,132 @@ func TestCadenceArch(t *testing.T) {
 	})
 }
 
+func TestEVMDeployFileSystemContract(t *testing.T) {
+	chain := flow.Emulator.Chain()
+
+	RunContractWithNewEnvironment(
+		t,
+		chain,
+		GetFileSystemContract(t),
+		func(
+			ctx fvm.Context,
+			vm fvm.VM,
+			snapshot snapshot.SnapshotTree,
+			testContract *TestContract,
+			testAccount *EOATestAccount,
+		) {
+			sc := systemcontracts.SystemContractsForChain(chain.ChainID())
+
+			code := []byte(fmt.Sprintf(
+				`
+					import EVM from %s
+
+					transaction(tx: [UInt8], coinbaseBytes: [UInt8; 20]){
+						prepare(account: &Account) {
+							let coinbase = EVM.EVMAddress(bytes: coinbaseBytes)
+							let res = EVM.run(tx: tx, coinbase: coinbase)
+
+							assert(res.status == EVM.Status.successful, message: "unexpected status")
+							assert(res.errorCode == 0, message: "unexpected error code")
+							assert(res.deployedContract == nil, message: "unexpected deployed contract")
+						}
+					}
+					`,
+				sc.EVMContract.Address.HexWithPrefix(),
+			))
+
+			coinbaseAddr := types.Address{1, 2, 3}
+			coinbaseBalance := getEVMAccountBalance(t, ctx, vm, snapshot, coinbaseAddr)
+			require.Zero(t, types.BalanceToBigInt(coinbaseBalance).Uint64())
+
+			var buffer bytes.Buffer
+			address := common.HexToAddress("0xea02F564664A477286B93712829180be4764fAe2")
+			chunkHash := "0x2521660d04da85198d1cc71c20a69b7e875ebbf1682f6a5c6a3fec69068ccc13"
+			index := int64(0)
+			chunk := "T1ARYBYsPOJPVYoU7wVp+PYuXmdS6bgkLf2egcxa+1hP64wAxfkLXtnllU6DmEuj+Id4oWl1ZV4ftQ+ofQ3DQhOoxNlPZGOYbhoMLuzE"
+			for i := 0; i <= 500; i++ {
```

Comment:
By increasing the chunk length, we can simulate EVM transactions with higher and higher gas consumption. The gas usage mainly comes from read/write storage operations.